### ISSUE TITLE
cmd/doc: go doc rsc.io/quote/v3 in module mode shows wrong import path

### ISSUE THREAD

**Issue by rsc** [golang/go#30778]
In module mode, in a module with rsc.io/quote/v3 as a dependency:

```
$ go doc rsc.io/quote/v3
package quote // import "rsc.io/quote"

Package quote collects pithy sayings.

func Concurrency() string
func GlassV3() string
func GoV3() string
func HelloV3() string
func OptV3() string
$ 
```

Note the import comment on the first line of output.
It should say `rsc.io/quote/v3`.
I do not know why it says `rsc.io/quote` instead.

**Comment by go101** [comment]
It looks the specified import path doesn't work when modules feature is enabled.

**Comment by dmitshur** [comment]
I'll be able to look into this issue.

**Comment by dmitshur** [comment]
Actually, I can't reproduce this with Go 1.12.1 nor Go tip (at commit fd270d8b579cc9dd4ced34fa7605a444794a787d):

```
$ cd $(mktemp -d)
$ go mod init m
go: creating new go.mod: module m
$ go get rsc.io/quote/v3
$ go doc rsc.io/quote/v3
package quote // import "rsc.io/quote/v3"

Package quote collects pithy sayings.

func Concurrency() string
func GlassV3() string
func GoV3() string
func HelloV3() string
func OptV3() string
```

@rsc What version of Go did you use? Where were the commands you ran before to initialize the main module?

**Comment by douglarek** [comment]
```
~/w/g/hello $ go doc rsc.io/quote/v3
package quote // import "rsc.io/quote/v3"

Package quote collects pithy sayings.

func Concurrency() string
func GlassV3() string
func GoV3() string
func HelloV3() string
func OptV3() string
~/w/g/hello $ go get rsc.io/quote/v3
~/w/g/hello $ go doc rsc.io/quote/v3
package quote // import "rsc.io/quote/v3"

Package quote collects pithy sayings.

func Concurrency() string
func GlassV3() string
func GoV3() string
func HelloV3() string
func OptV3() string
```

**Comment by dmitshur** [comment]
There's a chance this is the same issue as https://github.com/golang/go/issues/28992#issuecomment-483441942.

**Comment by andybons** [comment]
@rsc is this still an issue? Do you have updated repro steps? Thanks.

**Comment by mseshachalam** [comment]
@dmitshur 
```
$ go doc rsc.io/quote/v3
package quote // import "rsc.io/quote"

Package quote collects pithy sayings.

func HelloV3() string
```
I ran this on go version go1.12.6 linux/amd64 in a local module aftering `go get rsc.io/quote/v3`

**Comment by dmitshur** [comment]
@mseshachalam Did you have a copy of `rsc.io/quote` repository in your GOPATH/src?

Can you check if commit 3b040b7e8088ad2c02e413a4abf7effcd62373d0 fixes it?

**Comment by mseshachalam** [comment]
@dmitshur I have a copy of `rsc.io/quote' at $GOPATH/src/rsc.io/quote pointing master branch at https://github.com/rsc/quote/commit/5d9f230bcfbae514bb6c2215694c2ce7273fc604 
I had to correct a syntax error at v3/quote.go
```
git diff v3/quote.go
diff --git a/v3/quote.go b/v3/quote.go
index 1f5f3d6..9b6071f 100644
--- a/v3/quote.go
+++ b/v3/quote.go
@@ -10,3 +10,4 @@ import "rsc.io/sampler"
 // Hello returns a greeting.
 func HelloV3() string {
        return sampler.Hello()
+}
```

**Comment by dmitshur** [comment]
Go 1.12.7 does not include it, it's currently only available on the latest `master` branch. You can use [`golang.org/dl/gotip`](https://godoc.org/golang.org/dl/gotip) to try it.

**Comment by mseshachalam** [comment]
Yeah. `gotip` works as expected.

```
seshachalamm@IMCHLT047:~/hello$ gotip doc rsc.io/quote/v3
package quote // import "rsc.io/quote/v3"

Package quote collects pithy sayings.

func Concurrency() string
func GlassV3() string
func GoV3() string
func HelloV3() string
func OptV3() string
```
```
seshachalamm@IMCHLT047:~/hello$ go doc rsc.io/quote/v3
package quote // import "rsc.io/quote"

Package quote collects pithy sayings.

func HelloV3() string
```

Could you please explain briefly about the changes in gotip and go 1.12.7 ? The output is significantly different.

**Comment by dmitshur** [comment]
Thanks for confirming @mseshachalam.

The change was [CL 183991](https://golang.org/cl/183991), you can read its commit message for details. It fixed issue #28992, which was that in some conditions, `go doc` was incorrectly using packages in your GOPATH workspace instead of from the appropriate module.

> I imagine gotip is using the tags of a library and go1.12.6 is using master branch ?

Yes, it's now using the `rsc.io/quote/v3` module (at some version as determined by your main module), instead of the `v3` subdirectory of `rsc.io/quote` repository in your GOPATH workspace (on `master` branch).
